### PR TITLE
Add icon size prop

### PR DESCRIPTION
**Pull request by mrazian85**
[Link to updated components in Deploy Preview](https://deploy-preview-150--neo-react-library-storybook.netlify.app/?path=/story/components-icon--standard-icon-sm)

**Before tagging the team for a review, I have done the following:**
- [x] reviewed my code changes
- [x] ensured that all tests pass
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] tagged Matt if any visual changes have occurred
- [x] done an accessibility check on my work

Adding logic to toggle icon size class names as per the latest additions to the [Neo CSS library](https://design.avaya.com/whats-new/changelog).

Updated to latest Neo CSS version as part of the above work. Also fixed a failing test that relied on Icon Component snapshots.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,7 +68,7 @@
     "react-table": "7.8.0"
   },
   "devDependencies": {
-    "@avaya/neo": "3.67.0",
+    "@avaya/neo": "^3.68.0",
```
Comment (joe-s-avaya) [inline package.json:71]: revert the adding of the carrot (`^`)

File: src/components/Icon/Icon.tsx [hunk 2]
```diff
@@ -75,7 +76,7 @@ export const Icon: React.FC<IconProps> = ({
       result.push(iconClass);
     }
 
-    return result.join(" ");
+    return result.join(" ").trimStart();
```
Comment (joe-s-avaya) [inline src/components/Icon/Icon.tsx:79]: just use `trim` no reason to only get one end
```suggestion
    return result.join(" ").trim();
```

Comment (joe-s-avaya) [reply src/components/Icon/Icon.tsx:79]: for bonus points, convert this using a `switch` to using `clsx` (which is what we should have done from the start 😬, but hindsight is 20-20)

File: package.json (hunk 1 above)
Comment (mrazian85) [reply package.json]: ✅

File: src/components/Icon/Icon.tsx (hunk 2 above)
Comment (mrazian85) [reply src/components/Icon/Icon.tsx]: ✅

File: src/components/Icon/Icon.tsx [hunk 3]
```diff
@@ -1,9 +1,10 @@
+import clsx from "clsx";
 import { useMemo } from "react";
```
Comment (joe-s-avaya) [inline src/components/Icon/Icon.tsx:2]: Screenshot 2023-03-06 at 10 33 01 AM

**Review by joe-s-avaya — APPROVED** [review]
there's a linting issue, but otherwise all good

**Pull request merged**